admiral/pkg/clusters/serviceentry.go

```diff
@@ -824,7 +824,7 @@ func modifyServiceEntryForNewServiceOrPod(
 	// VS Based Routing
 	// Writing phase: We update the base ingress virtualservices with the RouteDestinations
 	// gathered during the discovery phase and write them to the source cluster
-	err = addUpdateVirtualServicesForIngress(ctx, ctxLogger, remoteRegistry, sourceClusterToDestinations)
+	err = addUpdateVirtualServicesForIngress(ctx, ctxLogger, remoteRegistry, sourceClusterToDestinations, cname)
```

Comment:
since a vs can have multiple hosts, would it be advisable to use an array of `cnames`?

Reply:
@adilfulara this is actually the vs name and I'm just reusing the cname thats being used in other places where cname variable name makes sense. I can create another variable or add a comments describing the intention of this variable in this context.